build/source/engine/groundwatr.f90

```diff
@@ -419,117 +328,81 @@ subroutine computeBaseflow(&
    trTotal(iLayer) = tran0*(zActive(iLayer)/soilDepth)**zScale_TOPMODEL
    trSoil(iLayer)  = trTotal(iLayer) - trTotal(iLayer+1)
   end if
-  !write(*,'(a,1x,i4,1x,10(f20.15,1x))') 'iLayer, mLayerMatricHeadLiq(iLayer), mLayerVolFracLiq(iLayer), zActive(iLayer), trTotal(iLayer), trSoil(iLayer) = ', &
-  !                                       iLayer, mLayerMatricHeadLiq(iLayer), mLayerVolFracLiq(iLayer), zActive(iLayer), trTotal(iLayer), trSoil(iLayer)
- end do  ! looping through soil layers
+ end do  ! end looping through soil layers
 
  ! set un-used portions of the vectors to zero
- if(ixSaturation>1)then
+ if (ixSaturation>1) then
   zActive(1:ixSaturation-1) = 0._rkind
   trTotal(1:ixSaturation-1) = 0._rkind
   trSoil(1:ixSaturation-1)  = 0._rkind
  end if
 
- ! compute the outflow from each layer (m3 s-1)
- mLayerColumnOutflow(1:nSoil) = trSoil(1:nSoil)*tan_slope*contourLength
+ mLayerColumnOutflow(1:nSoil) = trSoil(1:nSoil)*tan_slope*contourLength ! compute the outflow from each layer (m3 s-1)
 
  ! compute total column inflow and total column outflow (m s-1)
  totalColumnInflow  = sum(mLayerColumnInflow(1:nSoil))/HRUarea
  totalColumnOutflow = sum(mLayerColumnOutflow(1:nSoil))/HRUarea
 
- ! compute the available storage (m)
- availStorage = sum(mLayerDepth(1:nSoil)*(theta_sat - (mLayerVolFracLiq(1:nSoil)+mLayerVolFracIce(1:nSoil))) )
+ availStorage = sum(mLayerDepth(1:nSoil)*(theta_sat - (mLayerVolFracLiq(1:nSoil)+mLayerVolFracIce(1:nSoil)))) ! compute the available storage (m)
 
  ! compute the smoothing function (-)
- if(availStorage < xMinEval)then
-  ! (compute the logistic function)
+ if (availStorage < xMinEval) then
+  ! compute the logistic function
   expF = exp((availStorage - xCenter)/xWidth)
   logF = 1._rkind / (1._rkind + expF)
   ! (compute the derivative in the logistic function w.r.t. volumetric liquid water content in each soil layer)
-  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2._rkind
+  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2_i4b !!integer exponent
  else
   logF             = 0._rkind
   dLogFunc_dLiq(:) = 0._rkind
  end if
 
- ! compute the exfiltartion (m s-1)
+ ! compute the exfiltration (m s-1)
  if(totalColumnInflow > totalColumnOutflow .and. logF > tiny(1._rkind))then
   scalarExfiltration = logF*(totalColumnInflow - totalColumnOutflow)  ! m s-1
-  !write(*,'(a,1x,10(f30.20,1x))') 'scalarExfiltration = ', scalarExfiltration
  else
   scalarExfiltration = 0._rkind
  end if
 
- ! check
```

Comment:
@martynpclark Can you give an opinion on whether it's worth keeping (informal) debugging statements like these?

Reply:
This change was confirmed with @martynpclark in a meeting on June 6th.